### PR TITLE
Add typing for tmt/steps/execute/__init__.py

### PR DESCRIPTION
**Pull request by janhavlin**

### CODE REVIEW

File: tmt/steps/execute/__init__.py
Comment (janhavlin) [inline tmt/steps/execute/__init__.py:229]: As discussed in [#1319](https://github.com/teemtee/tmt/pull/1319#discussion_r917014429), this should return `List[ClickOptionDecoratorType]` which is being added in #1304. So blocking this PR until #1304 is merged.

File: tmt/steps/execute/__init__.py
Comment (janhavlin) [inline tmt/steps/execute/__init__.py:177]: Also blocked until #1319 is merged.

File: tmt/steps/execute/__init__.py
Comment (janhavlin) [reply tmt/steps/execute/__init__.py]: Resolved.

File: tmt/steps/execute/__init__.py
Comment (janhavlin) [reply tmt/steps/execute/__init__.py]: Resolved.

**Review by optak — APPROVED** [review]
LGTM

File: tmt/steps/execute/__init__.py
Comment (janhavlin) [inline tmt/steps/execute/__init__.py:235]: We should try to get rid of this cast(). PR https://github.com/teemtee/tmt/pull/1324 has https://github.com/teemtee/tmt/pull/1324#discussion_r926567419 on how to do that, so I'll wait if it gets approved before changing it here also.

File: tmt/steps/execute/__init__.py
Comment (psss) [inline tmt/steps/execute/__init__.py:61]: I believe here the `self.scripts` is misplaced, I guess that in 8fcd656c @thrix meant to add this to `ExecutePlugin` not the `Execute` step class.

File: tmt/steps/execute/__init__.py
Comment (psss) [inline tmt/steps/execute/__init__.py:206]: The number of scripts can vary, so this rather be the following?
```suggestion
    scripts: Tuple['Script', ...] = ()
```

File: tmt/steps/execute/__init__.py
Comment (psss) [inline tmt/steps/execute/__init__.py:217]: Should not be needed, `usage` is now mandatory.

File: tmt/steps/execute/__init__.py
Comment (psss) [inline tmt/steps/execute/__init__.py:226]: Should be `context: click.Context` instead?

File: tmt/steps/execute/__init__.py
Comment (psss) [inline tmt/steps/execute/__init__.py:299]: Should not be needed if we `scripts` is not `Optional` any more.

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for preparing this! Added a couple of comments.

File: tmt/steps/execute/__init__.py
Comment (thrix) [reply tmt/steps/execute/__init__.py:61]: +1, seems this was a mistake

Comment (janhavlin) [reply tmt/steps/execute/__init__.py]: Removed.

File: tmt/steps/execute/__init__.py
Comment (janhavlin) [reply tmt/steps/execute/__init__.py]: Updated, thanks!

File: tmt/steps/execute/__init__.py
Comment (thrix) [inline tmt/steps/execute/__init__.py:342]: as discussed in IRC, let's revisit this, we are trying to avoid type ignores

**Review by thrix — CHANGES_REQUESTED** [review]

File: tmt/steps/execute/__init__.py
Comment (thrix) [inline tmt/steps/execute/__init__.py:89]: I wonder, what is the problem here. I would expect `self.data` to be `tmt.steps.StepData`, which is

```
   71 class StepData(TypedDict, total=False):
   72     """
   73     Step data structure
   74     """
   75     name: str
   76     how: str
   77     order: Optional[int]
```

So why we need to assert it here?

Comment (thrix) [reply tmt/steps/execute/__init__.py]: assert was removed, thanks, resolving

**Review by thrix — APPROVED** [review]

**Comment by thrix** [comment]
@psss looks good to me to be merged, all problems seem to be addressed well.

**Review by psss — APPROVED** [review]
Thanks for all the fixes! Looks good now.

**Pull request merged**
